spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -50,32 +52,99 @@ public class GravitinoCatalog implements TableCatalog, SupportsNamespaces {
   // The Gravitino catalog client to do schema operations.
   protected Catalog gravitinoCatalogClient;
+  protected PropertiesConverter propertiesConverter;
 
   private final String metalakeName;
   private String catalogName;
   private final GravitinoCatalogManager gravitinoCatalogManager;
+  // Different catalog use GravitinoCatalogAdaptor to adapt to GravitinoCatalog
+  private GravitinoCatalogAdaptor gravitinoAdaptor;
 
   public GravitinoCatalog() {
     gravitinoCatalogManager = GravitinoCatalogManager.get();
     metalakeName = gravitinoCatalogManager.getMetalakeName();
   }
 
+  @Override
+  public void initialize(String name, CaseInsensitiveStringMap options) {
+    this.catalogName = name;
+    this.gravitinoCatalogClient = gravitinoCatalogManager.getGravitinoCatalogInfo(name);
+    String provider = gravitinoCatalogClient.provider();
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(provider), name + " catalog provider is empty");
+    this.gravitinoAdaptor = GravitinoCatalogAdaptorFactory.createGravitinoAdaptor(provider);
+    this.sparkCatalog =
+        gravitinoAdaptor.createAndInitSparkCatalog(
+            name, options, gravitinoCatalogClient.properties());
+    this.propertiesConverter = gravitinoAdaptor.getPropertiesConverter();
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
   @Override
   public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
-    throw new NotSupportedException("Doesn't support listing table");
+    String gravitinoNamespace;
+    if (namespace.length == 0) {
+      gravitinoNamespace = getCatalogDefaultNamespace();
+    } else {
+      validateNamespace(namespace);
+      gravitinoNamespace = namespace[0];
+    }
+    NameIdentifier[] identifiers =
+        gravitinoCatalogClient
+            .asTableCatalog()
+            .listTables(Namespace.of(metalakeName, catalogName, gravitinoNamespace));
+    return Arrays.stream(identifiers)
+        .map(identifier -> Identifier.of(getNamespace(identifier), identifier.name()))
+        .toArray(Identifier[]::new);
   }
 
   @Override
   public Table createTable(
       Identifier ident, Column[] columns, Transform[] partitions, Map<String, String> properties)
       throws TableAlreadyExistsException, NoSuchNamespaceException {
-    throw new NotSupportedException("Doesn't support creating table");
+    checkNotPathIdentifier(ident, "createTable");
+    NameIdentifier nameIdentifier =
+        NameIdentifier.of(metalakeName, catalogName, getDatabase(ident), ident.name());
+    com.datastrato.gravitino.rel.Column[] gravitinoColumns =
+        Arrays.stream(columns)
+            .map(column -> createGravitinoColumn(column))
+            .toArray(com.datastrato.gravitino.rel.Column[]::new);
+
+    Map<String, String> gravitinoProperties =
+        propertiesConverter.toGravitinoTableProperties(properties);
+    String comment = gravitinoProperties.remove("comment");
```

Comment:
Removing this attribute and retrieving the 'comment' is too abrupt an implementation. can you give an easy to understand implementation

Reply:
Unlike trino store comment in table meta data, spark stores comment in properties, seems it's necessary to do the transform. any suggestions?

Reply:
add some description to related code.

Reply:
My suggestion is to first separate the handling of 'getComment' and 'remove unused attribute' for easier understanding. Secondly, extracting 'comment' from attributes and setting 'comment' into attributes should be managed within a single class.